src/Scanner.ts

```diff
@@ -1,52 +1,170 @@
-import { Token, TokenType } from './Token';
+import { Token, TokenType } from './Token'
 
-class ScannerError extends Error {};
+class ScannerError extends Error {}
 
 function scan(input: string): Token[] {
-	return new Scanner(input).scanTokens();
+  return new Scanner(input).scanTokens()
 }
 
 export class Scanner {
-	public index: number = 0;
-
-	constructor(
-		public input: string,
-	) {}
-
-	scanTokens(): Token[] {
-		const tokens = [];
-
-		while (!this.isAtEnd()) {
-			const token = this.scanNext();
-			if (token) {
-				tokens.push(token);
-			}
-		}
-
-		return tokens;
-	}
-
-	scanNext(): Token {
-		const char = this.advance();
-
-		if (char === "+") {
-			return {
-				ttype: TokenType.PLUS,
-				text: "+",
-			};
-		}
-		throw(new ScannerError(`scanNext: unrecognized character ${char}`));
-	}
-
-	isAtEnd(): boolean {
-		throw(new ScannerError("isAtEnd: not implemented"));
-	}
-
-	peek(): string {
-		throw(new ScannerError("peek: not implemented"));
-	}
-
-	advance(): string {
-		throw(new ScannerError("advance: not implemented"));
-	}
+  public index: number = 0
+
+  constructor(public input: string) {}
+
+  scanTokens(): Token[] {
+    const tokens = []
+    if (this.peek() == ' ') {
+      this.advance()
+    }
+
+    while (!this.isAtEnd()) {
+      const token = this.scanNext()
+      if (token) {
+        tokens.push(token)
+      }
+    }
+
+    return tokens
+  }
+
+  scanNext(): Token {
+    const char = this.advance()
+    switch (char) {
+      case '+':
+        return {
+          ttype: TokenType.PLUS,
+          text: '+',
+        }
+        break
+      case '-':
+        return {
+          ttype: TokenType.MINUS,
+          text: '-',
+        }
+        break
+      case '*':
+        return {
+          ttype: TokenType.TIMES,
+          text: '*',
+        }
+        break
+      case '/':
+        return {
+          ttype: TokenType.DIVIDE,
+          text: '/',
+        }
+        break
+      case '(':
+        return {
+          ttype: TokenType.LEFT_PAREN,
+          text: '(',
+        }
+        break
+      case ')':
+        return {
+          ttype: TokenType.RIGHT_PAREN,
+          text: ')',
+        }
+        break
+      case '^':
+        return {
+          ttype: TokenType.POW,
+          text: '^',
+        }
+        break
+      case '0':
+        return {
+          ttype: TokenType.NUMBER,
+          text: '0',
+          literal: 0,
+        }
+        break
+      case '1':
```

Comment:
Instead of having all these case statement for numeric values - you can use the fallthrough nature of case statements to have one return for all of them:

```javascript
  case '0':
  case '1':
  ...
  case '9':
     return {
       ttype: TokenType.NUMBER,
       text: char,
       literal: parseFloat(char),
     }

```

Reply:
Got the cases simplified so they fallthrough. Tests pass successfully.